iothub_client/tests/common_e2e/iothubclient_common_e2e.c

```diff
@@ -681,56 +843,86 @@ D2C_MESSAGE_HANDLE client_create_and_send_d2c(TEST_MESSAGE_CREATION_MECHANISM te
         }
         ASSERT_IS_NOT_NULL(msgHandle, "Could not create the D2C message to be sent");
 
-        for (size_t i = 0; i < MSG_PROP_COUNT; i++)
+        for (size_t j = 0; j < MSG_PROP_COUNT; j++)
         {
             if (g_e2e_test_options.use_special_chars)
             {
-                if (IoTHubMessage_SetProperty(msgHandle, MSG_PROP_KEYS_SPECIAL[i], MSG_PROP_VALS_SPECIAL[i]) != IOTHUB_MESSAGE_OK)
+                if (IoTHubMessage_SetProperty(msgHandle, MSG_PROP_KEYS_SPECIAL[j], MSG_PROP_VALS_SPECIAL[j]) != IOTHUB_MESSAGE_OK)
```

Comment:
I get this is pre-existing condition, but... shouldn't a `IoTHubMessage_SetProperty` cause an ASSERT_FAIL.  It suggets something is really wrong if something this basic fails and also later in test we'll see if these are set and fail there if not.  I'd rather stop test at root problem.

Reply:
I will focus on the goal of the changes proposed for AMQP max message size.
Here is a github issue filed to track this request: https://github.com/Azure/azure-iot-sdk-c/issues/2360